src/Campaigns/Migrations/Tables/CreateCampaignsTable.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Give\Campaigns\Migrations\Tables;
+
+use Give\Framework\Database\DB;
+use Give\Framework\Database\Exceptions\DatabaseQueryException;
+use Give\Framework\Migrations\Contracts\Migration;
+use Give\Framework\Migrations\Exceptions\DatabaseMigrationException;
+
+/**
+ * @unreleased
+ * Creates give_campaigns table
+ */
+class CreateCampaignsTable extends Migration
+{
+    /**
+     * @inheritdoc
+     */
+    public static function id(): string
+    {
+        return 'give-campaigns-create-give-core-campaigns-table';
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public static function title(): string
+    {
+        return 'Create give_campaigns table from core';
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public static function timestamp(): string
+    {
+        return strtotime('2024-08-26 00:00:00');
+    }
+
+    /**
+     * @inheritDoc
+     * @throws DatabaseMigrationException
+     */
+    public function run()
+    {
+        global $wpdb;
+
+        $table = $wpdb->prefix . 'give_campaigns';
+        $charset = DB::get_charset_collate();
+
+        $sql = "CREATE TABLE $table (
+			id INT UNSIGNED NOT NULL AUTO_INCREMENT,
+			form_id INT NOT NULL,
+			campaign_type VARCHAR(12) NOT NULL DEFAULT '',
```

Comment:
We'll need to add a one-time migration that sets the `campaign_type` for preexisting P2P campaigns.

Reply:
Resolved in another PR: https://github.com/impress-org/givewp/pull/7520/commits/d52f7697bd6b1383bf98eaf2b0aa3e6ca1272f6a